src/diffusers/models/vae.py

```diff
@@ -280,6 +280,292 @@ def custom_forward(*inputs):
         return sample
 
 
+class UpSample(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        activate_before: str = "none",
+        activate_after: str = "none",
+        upsample_type: str = "deconv",
+    ) -> None:
+        super().__init__()
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.activate_before = activate_before
+        self.activate_after = activate_after
+        self.upsample_type = upsample_type
+        if self.upsample_type == "deconv":
+            self.deconv = nn.ConvTranspose2d(in_channels, out_channels, kernel_size=4, stride=2, padding=1)
+        else:
+            if self.upsample_type in ["bilinear", "nearest"]:
+                self.conv = nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=1, padding=1)
+            else:
+                raise NotImplementedError
+
+    def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
+        if self.activate_before == "relu":
+            x = torch.relu(x)
+        elif self.activate_before == "none":
+            pass
+        else:
+            raise NotImplementedError
+
+        if self.upsample_type == "deconv":
+            x = self.deconv(x)
+        else:
+            x = nn.functional.interpolate(x, scale_factor=2.0, mode=self.upsample_type)
+            x = self.conv(x)
+
+        if self.activate_after == "relu":
+            x = torch.relu(x)
+        elif self.activate_after == "none":
+            pass
+        else:
+            raise NotImplementedError
+
+        return x
+
+
+class ConditionEncoder(nn.Module):
```

Comment:
```suggestion
class MaskConditionEncoder(nn.Module):
```

Maybe make it explicit that it's for masked images and the mask?